dream_demo.ipynb

```diff
@@ -0,0 +1,261 @@
+    "\n",
+    "The wavelength window can be adjusted. \n",
+    "\n",
+    "A simple text based progress bar will be shown unless the keyword argument `verbose` is set to False. \n",
+    "\n",
+    "The resulting data is checked and compared to the ESS timeframe, it would be optimal to have just under 100% coverage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Generating random sparse data\")\n",
+    "Dream.generate_data_pseudo(n_events, wavelength_width=3.5, wavelength_center=3.0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(Dream.data) # Print an overview of the resulting dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert time of flight data to dspacing\n",
+    "The raw time of flight data now needs to be converted back to dspacing. This can be done using the calibration data calculated within the Dream object, or without assuming a perfect instrument. \n",
+    "\n",
+    "The two methods will give the same results in this case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dspacing = powder_reduction.powder_reduction(Dream.data, calibration=Dream.cal)\n",
+    "print(dspacing) # Show the converted dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(dspacing.dims) # Show the dimensionality of the dspacing dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plotting the dspacing dataset\n",
+    "First the raw data is collected from each of the sparse channels in the detector array. Next `numpy` is used to histogram the data, after which the data is plotted with `matplotlib`. We see a diffraction pattern with peaks in the specified interval."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract the sparse dspacing data from the scipp variable\n",
+    "raw_data = []\n",
+    "for i in range(len(dspacing.coords[Dim.Position].values)):\n",
+    "    sparse = dspacing['sample'].coords[Dim.DSpacing][Dim.Position, i].values\n",
+    "    this_data = [sparse[k] for k in range(len(sparse))]\n",
+    "    raw_data.extend(this_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot using scipp feature\n",
+    "import numpy as np\n",
+    "from scipp.plot import plot\n",
+    "%matplotlib inline\n",
+    "\n",
+    "data_to_plot = sc.Dataset()\n",
+    "N = 501\n",
+    "bins = np.linspace(2.0, 10.0, N).astype(np.float64)\n",
```

Comment:
Since yesterday, we can plot sparse data directly (binning internally), see https://scipp.readthedocs.io/en/latest/user-guide/plotting.html#Plotting-sparse-data.